extmod/modbluetooth_nimble.c

```diff
@@ -353,10 +365,13 @@ void mp_bluetooth_get_device_addr(uint8_t *addr) {
 }
 
 int mp_bluetooth_gap_advertise_start(bool connectable, int32_t interval_us, const uint8_t *adv_data, size_t adv_data_len, const uint8_t *sr_data, size_t sr_data_len) {
-    int ret;
+    if (!mp_bluetooth_is_active()) {
+        return MP_EINVAL;
+    }
```

Comment:
How about using a specific error code to signal this-is-not-initialized? This would allow us to report a specific error message. This is what I would have proposed: https://github.com/codefreax/micropython/commit/73d7640e11c98042ee3b353de3aef92576e63844

Reply:
@codefreax yes I think using a more specific error code is a good idea, makes it easier to debug problems.  I see you suggested `EFAULT`, but I think that's mainly for memory/bus faults.  So I'd suggest either `EPERM` (operation not permitted), `EIO` or `ENODEV`.